client/transaction-pool/README.md

```diff
@@ -1,3 +1,349 @@
 Substrate transaction pool implementation.
 
-License: GPL-3.0-or-later WITH Classpath-exception-2.0
\ No newline at end of file
+License: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+# Problem Statement
+
+The transaction pool is responsible for maintaining a set of transactions that
+possible to include by block authors in upcoming blocks. Transactions are received
+either from networking (gossiped by other peers) or RPC (submitted locally).
```

Comment:
Maybe it should mention that when receiving transactions with the same bytes multiple times, it is only included once in the pool.

Reply:
> Actually this is something that we deduct indirectly from provides tag description.

That's an interest take on this, but it implies that we should add the same transaction multiple times in the pool in case the `provides` tag is empty.

But this raises lots of question: if you receive the same transaction twice from the network, how do you know whether they're a unique transaction that's been gossiped around, or if it's actually two different people who have submitted the same transaction?

Reply:
> That's an interest take on this, but it implies that we should add the same transaction multiple times in the pool in case the provides tag is empty.

The `provides` tag list cannot be empty - such transactions should be considered invalid, we could panic in the runtime call actually if someone tries to do that, but the current pool implementation does not allow that.

>  if it's actually two different people who have submitted the same transaction?

If the transaction is signed, it will contain two different signatures, so it's not the exact same payload. If it's unsigned, then well, I'd argue that de-duplication is sensible, cause it's literally nothing that allows us to distinguish one transaction from the other (or identify the sender for instance).